nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -630,17 +629,30 @@ in
       }
     ];
 
+    security.wrappers.nginx = {
```

Comment:
@delroth 
You can rename to security.wrappers.test-nginx ?

Reply:
Done.

Reply:
Huh what's the reason behind this? This has nothing to do with testing

Reply:
"chown" must be ran before "${cfg.package}/bin/nginx ... -t", otherwise an error will be observed. More details in https://github.com/NixOS/nixpkgs/pull/48875#pullrequestreview-167403437

Reply:
@Infinisil it does have to do with testing, because this wrapper for the nginx binary is only used for the config testing done at pre-startup. The reason why it needs a wrapper is that for some reason the nginx config test checks if ports can be listened on, and this fails for privileged ports without the proper cap. In serving situations, that cap is given by the systemd service. In pre-exec the cap is not provided, hence the need for this wrapper (which is only runnable by user nginx, so doesn't provide a universal cap privesc).

It's not very elegant and probably a better way would be to convince upstream to have an option to not do this listen port check. But frankly this is more effort than I'm willing to put into this -- way more effort than is required to rebase this PR in my local fork to get the benefits of it myself.

Reply:
@delroth there is still such a method https://github.com/nixcloud/nixcloud-webservices/blob/master/modules/web/webserver/lib/nginx_check_config.nix